### ISSUE TITLE
Poetry in txt to HTML conversion

### ISSUE THREAD

**Issue by gbnewby** [gutenbergtools/ebookmaker#26]
Discussed among WWers and catalogers: This is for files that are native .txt, and get automatically converted to HTML as part of ebookmaker.

The problem is that spacing in poetry gets ignored. The easiest partial solution is to insert <br /> or similar whenever there is a space at the start of a line.

Problematic generated HTML may be seen, for example, here:
https://www.gutenberg.org/ebooks/60909
https://www.gutenberg.org/ebooks/60910

Compare the .txt to the generated .htm. When a new line begins with upper case, it's treated as a paragraph. When a new line begins with lower case, it's treated as part of the previous paragraph.

Email pgww (or gutcat) if you want further examples or to discuss solutions. Thanks for considering this.

**Comment by eshellman** [comment]
I can't think of any way to fix these without breaking conversion of thousands of non poetry files.
workarounds include:

- wrapping the poetry in html by hand with `<pre>`
- using rst markup instead of plain text
- inventing new markup
- magic

**Comment by eshellman** [comment]
We've seen a lot of problems with poetry in html books where `<br>` is used in conjunction with a css rule hiding `br` in a block `span`. kindlegen can't handle more than 10K hidden characters.

**Comment by Rossolson** [comment]
Is there a list of the files that are showing the problem? I imagine there is a finite number (though perhaps large) of files in this situation? And relatively static and not growing now?

Perhaps there is a database query that can generate the list of "text-only poetry" works?

**Comment by eshellman** [comment]
No, that is a guiguts convention. I don't think PG gets txt only submissions from guiguts users.

**Comment by tangledhelix** [comment]
I would expect that you're correct, at least for DP because HTML is normally expected. What I was saying is I was told (long ago now) that PG may choose to rewrap the file later, and this was a convention to avoid improper rewrapping. It may have been more myth than fact.

**Comment by srjfoo** [comment]
> Is there a list of the files that are showing the problem? I imagine there is a finite number (though perhaps large) of files in this situation? And relatively static and not growing now?
> 
> Perhaps there is a database query that can generate the list of "text-only poetry" works?

I don't know how many other providers nowadays provide text only, but as far as I am aware, most of the Finnish books are still text only, with no supplied cover images.
